recirq/qcqmc/optimize_wf_test.py

```diff
@@ -0,0 +1,391 @@
+
+
+def test_pp_wf_energy(fixture_4_qubit_ham: HamiltonianData):
+    params = PerfectPairingPlusTrialWavefunctionParams(
+        name="pp_test_wf_1",
+        hamiltonian_params=fixture_4_qubit_ham.params,
+        heuristic_layers=(),
+        do_pp=True,
+        restricted=True,
+    )
+
+    trial_wf = build_pp_plus_trial_wavefunction(
+        params, dependencies={fixture_4_qubit_ham.params: fixture_4_qubit_ham}
+    )
+
+    assert np.isclose(trial_wf.ansatz_energy, fixture_4_qubit_ham.e_fci)
+
+
+def test_pp_wf_energy_with_layer(fixture_4_qubit_ham: HamiltonianData):
+    params = PerfectPairingPlusTrialWavefunctionParams(
+        name="pp_test_wf_2",
+        hamiltonian_params=fixture_4_qubit_ham.params,
+        heuristic_layers=(LayerSpec("charge_charge", "cross_spin"),),
+        do_pp=True,
+        restricted=True,
+    )
+
+    trial_wf = build_pp_plus_trial_wavefunction(
+        params, dependencies={fixture_4_qubit_ham.params: fixture_4_qubit_ham}
+    )
+
+    assert np.isclose(trial_wf.ansatz_energy, fixture_4_qubit_ham.e_fci)
+
+
+def test_qchem_pp_eight_qubit_wavefunctions_consistent(
+    fixture_8_qubit_ham: HamiltonianData,
+):
+    """Tests (without optimization) that the eight qubit wavefunctions work.
+
+    Specifically, that constructing the wavefunction with FQE and then
+    converting it to a cirq wavefunction yields the same result as constructing
+    the parameters with the circuit directly.
+    """
+    params = PerfectPairingPlusTrialWavefunctionParams(
+        name="pp_test_wf_qchem",
+        hamiltonian_params=fixture_8_qubit_ham.params,
+        heuristic_layers=tuple(),
+        initial_orbital_rotation=None,
+        initial_two_body_qchem_amplitudes=np.asarray([0.3, 0.4]),
+        do_optimization=False,
+    )
+
+    trial_wf = build_pp_plus_trial_wavefunction(
+        params,
+        dependencies={fixture_8_qubit_ham.params: fixture_8_qubit_ham},
+        do_print=False,
+    )
+
+    one_body_params = trial_wf.one_body_params
+    two_body_params = trial_wf.two_body_params
+    basis_change_mat = trial_wf.one_body_basis_change_mat
+
+    np.testing.assert_array_almost_equal(one_body_params, np.zeros((12,)))
+    np.testing.assert_array_almost_equal(basis_change_mat, np.diag(np.ones(8)))
+
+    np.testing.assert_equal(two_body_params.shape, (2,))
+
+
+def test_pp_plus_wf_energy_sloppy_1(fixture_8_qubit_ham: HamiltonianData):
+    params = PerfectPairingPlusTrialWavefunctionParams(
+        "pp_plus_test",
+        hamiltonian_params=fixture_8_qubit_ham.params,
+        heuristic_layers=tuple(),
+        do_pp=True,
+        restricted=False,
+        random_parameter_scale=1,
+    )
+
+    trial_wf = build_pp_plus_trial_wavefunction(
+        params,
+        dependencies={fixture_8_qubit_ham.params: fixture_8_qubit_ham},
+        do_print=True,
+    )
+
+    assert trial_wf.ansatz_energy < -1.947
+
+
+# TODO: Speed up this test and add a similar one with non-trivial heuristic layers.
```

Comment:
@wjhuggins, do you remember where this is from? Running these tests only takes 7 seconds total which isn't necessarily slow, but I do see quite significant creep in the CI's duration (from 2m to 13! minutes) since these PRs started to go in. I'll follow up with a PR to make things faster.

Reply:
yeah somehow these are exceptionally slow for actions, but only take a few seconds on my laptop. I'll investigate